RecoTracker/MkFitCore/interface/HitStructures.h

```diff
@@ -0,0 +1,874 @@
+      for (int qb = q_bin_1; qb < q_bin_2; ++qb) {
+        empty_phi_bins_dead(qb, 0, Config::m_nphi);
+      }
+    }
+
+  public:
+    LayerOfHits() {}
+
+    ~LayerOfHits() {
+#ifdef COPY_SORTED_HITS
+      free_hits();
+#endif
+      operator delete[](m_hit_ranks);
+    }
+
+    void SetupLayer(const LayerInfo& li);
+
+    void Reset() {}
+
+    float NormalizeQ(float q) const { return std::clamp(q, m_qmin, m_qmax); }
+
+    int GetQBin(float q) const { return (q - m_qmin) * m_fq; }
+
+    int GetQBinChecked(float q) const { return std::clamp(GetQBin(q), 0, m_nq - 1); }
+
+    // if you don't pass phi in (-pi, +pi), mask away the upper bits using m_phi_mask or use the Checked version.
+    int GetPhiBinFine(float phi) const { return std::floor(m_fphi_fine * (phi + Config::PI)); }
+    int GetPhiBin(float phi) const { return GetPhiBinFine(phi) >> m_phi_bits_shift; }
+
+    int GetPhiBinChecked(float phi) const { return GetPhiBin(phi) & m_phi_mask; }
+
+    const vecPhiBinInfo_t& GetVecPhiBinInfo(float q) const { return m_phi_bin_infos[GetQBin(q)]; }
+
+    // Get in all hits from given hit-vec
+    void SuckInHits(const HitVec& hitv);
+
+    // Get in all hits from given dead-vec
+    void SuckInDeads(const DeadVec& deadv);
+
+    // Use external hit-vec and only use hits that are passed to me.
+    void BeginRegistrationOfHits(const HitVec& hitv);
+    void RegisterHit(int idx);
+    void EndRegistrationOfHits(bool build_original_to_internal_map);
+
+    // Use this to map original indices to sorted internal ones. m_ext_idcs needs to be initialized.
+    int GetHitIndexFromOriginal(int i) const { return m_ext_idcs[i - m_min_ext_idx]; }
+    // Use this to remap internal hit index to external one.
+    int GetOriginalHitIndex(int i) const { return m_hit_ranks[i]; }
+
+#ifdef COPY_SORTED_HITS
+    const Hit& GetHit(int i) const { return m_hits[i]; }
+    const Hit* GetHitArray() const { return m_hits; }
+#else
+    const Hit& GetHit(int i) const { return (*m_ext_hits)[i]; }
+    const Hit* GetHitArray() const { return m_ext_hits->data(); }
+#endif
+
+    // void  SelectHitIndices(float q, float phi, float dq, float dphi, std::vector<int>& idcs, bool isForSeeding=false, bool dump=false);
+
+    void PrintBins();
+  };
+
+  //==============================================================================
+
+  class EventOfHits {
+  public:
+    std::vector<LayerOfHits> m_layers_of_hits;
+    int m_n_layers;
+    BeamSpot m_beam_spot;
+
+  public:
+    EventOfHits(const TrackerInfo& trk_inf);
+
+    void Reset() {
+      for (auto& i : m_layers_of_hits) {
+        i.Reset();
+      }
+    }
+
+    void SuckInHits(int layer, const HitVec& hitv) {
+      m_layers_of_hits[layer].SuckInHits(hitv);
+      /*
+    int   nh  = hitv.size();
+    auto &loh = m_layers_of_hits[layer];
+    loh.BeginRegistrationOfHits(hitv);
+    for (int i = 0; i < nh; ++i) loh.RegisterHit(i);
+    loh.EndRegistrationOfHits();
+    */
```

Comment:
remove if not used

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/20e8958ee9ec6436b3be8a101b90d9f1b956f31c